contracts/core/RebalancingSet.sol

```diff
@@ -0,0 +1,389 @@
+ */
+
+
+contract RebalancingSet is
+    StandardToken,
+    DetailedERC20
+{
+    using SafeMath for uint256;
+
+    /* ============ Enums ============ */
+
+    enum State { Default, Proposal, Rebalance }
+
+    /* ============ State Variables ============ */
+
+    address public factory;
+    uint256 public naturalUnit = 1;
+    address public manager;
+    State public rebalanceState;
+
+    // State updated after every rebalance
+    address public currentSet;
+    uint256 public shareRatio;
+    uint256 public lastRebalanceTimestamp;
+
+    // State governing rebalance cycle
+    uint256 public proposalPeriod;
+    uint256 public rebalanceInterval;
+
+    // State to track proposal period
+    uint256 public proposalStartTime;
+
+    // State needed for auction/rebalance
+    uint256 public auctionStartTime;
+    address public rebalancingSet;
+    address public auctionLibrary;
+    uint256 public auctionPriceDivisor;
+    uint256 public auctionStartPrice;
+    uint256 public curveCoefficient;
+    address[] public combinedTokenArray;
+    uint256[] public combinedCurrentUnits;
+    uint256[] public combinedRebalanceUnits;
+    uint256 public remainingCurrentSets;
+    uint256 public rebalanceSetSupply;
+
+    /* ============ Events ============ */
+
+    event NewManagerAdded(
+        address newManager,
+        address oldManager
+    );
+
+    event RebalanceProposed(
+        address rebalancingSet,
+        address indexed auctionLibrary,
+        uint256 indexed proposalPeriodEndTime
+    );
+
+    event RebalanceStarted(
+        address oldSet,
+        address newSet
+    );
+
+    event RebalanceFinished(
+        address newSet,
+        uint256 newShareRatio
+    );
+
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor function for Rebalancing Set Token
+     *
+     *
+     * @param _factory                      The factory used to create the Rebalancing Set
+     * @param _manager                      The manager of the Rebalancing Set
+     * @param _initialSet                   The initial set that collateralizes the Rebalancing set
+     * @param _initialShareRatio            How much of a Set (in gWei) equals one share
+     * @param _proposalPeriod               Amount of time for users to inspect a rebalance proposal
+     * @param _rebalanceInterval            The minimum amount of time between rebalances
+     * @param _name                         The Rebalancing Set's name
+     * @param _symbol                       The Rebalancing Set's symbol
+     */
+
+    constructor(
+        address _factory,
+        address _manager,
+        address _initialSet,
+        uint256 _initialShareRatio,
+        uint256 _proposalPeriod,
+        uint256 _rebalanceInterval,
+        string _name,
+        string _symbol
+    )
+        public
+        DetailedERC20(_name, _symbol, 18)
+    {
+        // Require day long proposal period
+        require(_proposalPeriod > 86400);
```

Comment:
I think ethereum has a global variable / unit called `day`

https://solidity.readthedocs.io/en/v0.4.24/units-and-global-variables.html?highlight=day

Reply:
Cool yeah still not sure what I wanna set that at, but yeah that's a good thing to use.